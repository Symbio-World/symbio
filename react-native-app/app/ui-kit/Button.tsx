import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
