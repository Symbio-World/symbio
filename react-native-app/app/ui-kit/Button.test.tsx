import * as React from 'react'
import { render } from 'react-native-testing-library'
import { Button, ButtonStyle } from './Button'

describe('Button', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Button />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('render transparent', () => {
    const { toJSON } = render(<Button variant={ButtonStyle.TRANSPARENT} />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('render fill', () => {
    const { toJSON } = render(<Button variant={ButtonStyle.FILL} />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('render outline', () => {
    const { toJSON } = render(<Button variant={ButtonStyle.OUTLINE} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
