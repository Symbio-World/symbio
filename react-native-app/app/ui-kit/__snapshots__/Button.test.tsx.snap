// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B35340",
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#fff",
        },
        Array [
          Object {
            "color": "#fff",
          },
        ],
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Button renders fill variant 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B35340",
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#fff",
        },
        Array [
          Object {
            "color": "#fff",
          },
        ],
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Button renders outline variant 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#B35340",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#fff",
        },
        Array [
          Object {
            "color": "#B35340",
          },
        ],
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Button renders title 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B35340",
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#fff",
        },
        Array [
          Object {
            "color": "#fff",
          },
        ],
      ]
    }
  >
    Button Test
  </Text>
</View>
`;

exports[`Button renders transparent variant 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0)",
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#fff",
        },
        Array [
          Object {
            "color": "#B35340",
          },
        ],
      ]
    }
  >
    
  </Text>
</View>
`;
